packages/server/integrations/JiraServerTaskIntegrationManager.ts

```diff
@@ -0,0 +1,108 @@
+import JiraServerIssueId from '~/shared/gqlIds/JiraServerIssueId'
+import {CreateTaskResponse, TaskIntegrationManager} from './TaskIntegrationManagerFactory'
+import JiraServerRestManager from './jiraServer/JiraServerRestManager'
+import {IGetTeamMemberIntegrationAuthQueryResult} from '../postgres/queries/generated/getTeamMemberIntegrationAuthQuery'
+import {TIntegrationProvider} from '../postgres/queries/getIntegrationProvidersByIds'
+import splitDraftContent from '~/utils/draftjs/splitDraftContent'
+import {ExternalLinks} from '~/types/constEnums'
+import IntegrationRepoId from '~/shared/gqlIds/IntegrationRepoId'
+
+export default class JiraServerTaskIntegrationManager implements TaskIntegrationManager {
+  public title = 'Jira Server'
+  private readonly auth: IGetTeamMemberIntegrationAuthQueryResult
+  private readonly provider: TIntegrationProvider
+
+  constructor(auth: IGetTeamMemberIntegrationAuthQueryResult, provider: TIntegrationProvider) {
+    this.auth = auth
+    this.provider = provider
+  }
+
+  private getManager() {
+    const {serverBaseUrl, consumerKey, consumerSecret} = this.provider
+    const {accessToken, accessTokenSecret} = this.auth
+    if (!serverBaseUrl || !consumerKey || !consumerSecret || !accessToken || !accessTokenSecret) {
+      throw new Error('Provider is not configured')
+    }
+    return new JiraServerRestManager(
+      serverBaseUrl,
+      consumerKey,
+      consumerSecret,
+      accessToken,
+      accessTokenSecret
+    )
+  }
+
+  async createTask({
+    rawContentStr,
+    integrationRepoId
+  }: {
+    rawContentStr: string
+    integrationRepoId: string
+  }): Promise<CreateTaskResponse> {
+    const manager = this.getManager()
+
+    const {title: summary, contentState} = splitDraftContent(rawContentStr)
+    // TODO: implement stateToJiraServerFormat
+    const description = contentState.getPlainText()
+
+    const {repoId, providerId} = IntegrationRepoId.split(integrationRepoId)
+
+    if (providerId !== this.provider.id || !repoId) {
+      throw new Error('Incorrect IntegrationRepoId')
```

Comment:
+1 we don't need to write unit tests or achieve 100% branch coverage, but if something is unreachable, then we probably don't need to write code for it.
Here, the repoId should be nonnull & we can trust the providerId from the split